shared/utility/tcp.hpp

```diff
@@ -73,39 +73,72 @@ inline int create_socket_server(const int& port) {
     address.sin_addr.s_addr = INADDR_ANY;
 
     // bind to port
-    if (bind(sfd, reinterpret_cast<sockaddr*>(&address), sizeof(sockaddr)) == -1) {
+    if (bind(server_fd, reinterpret_cast<sockaddr*>(&address), sizeof(sockaddr)) == -1) {
         std::cerr << "Cannot bind port " << port << std::endl;
-        close_socket(sfd);
+        close_socket(server_fd);
         return -1;
     }
 
     // listen for connections
-    if (listen(sfd, 1) == -1) {
+    if (listen(server_fd, 1) == -1) {
         std::cerr << "Cannot listen for connections" << std::endl;
-        close_socket(sfd);
+        close_socket(server_fd);
         return -1;
     }
     std::cerr << "Waiting for a connection on port " << port << " ..." << std::endl;
 
+    return server_fd;
+}
+
+inline int check_for_connection(const int& server_fd, const int& port) {
+
+    // Setup the polling data
+    pollfd fds;
+    fds.fd      = server_fd;
+    fds.events  = POLLIN | POLLPRI;  // Check for data to read and urgent data to read
+    fds.revents = 0;
+
+    // Poll the server fd to see if there is any data to read
+    const int num_ready = poll(&fds, 1, 0);
+
+    // Polling failed
+    if (num_ready < 0) {
+        std::cerr << "Error: Polling of TCP connection failed: " << strerror(errno) << std::endl;
+        return -1;
+    }
+
+    // We have an incoming connection
+    else if (num_ready > 0) {
 #ifdef _WIN32
-    int asize = sizeof(sockaddr_in);
+        int asize = sizeof(sockaddr_in);
 #else
-    socklen_t asize = sizeof(sockaddr_in);
+        socklen_t asize = sizeof(sockaddr_in);
 #endif
 
-    sockaddr_in client;
-    const int cfd = accept(sfd, reinterpret_cast<sockaddr*>(&client), &asize);
+        // Accept the connection
+        sockaddr_in client;
+        const int client_fd = accept(server_fd, reinterpret_cast<sockaddr*>(&client), &asize);
 
-    if (cfd == -1) {
-        std::cerr << "Cannot accept client" << std::endl;
-        close_socket(sfd);
-    }
-    else {
+        // Failed to accept the connection
+        if (client_fd == -1) {
+            std::cerr << "Error: Cannot accept client connection on port " << port << ": " << strerror(errno)
+                      << std::endl;
+            close_socket(server_fd);
+            return -1;
+        }
+
+        // Get client information
         const hostent* client_info = gethostbyname(inet_ntoa(client.sin_addr));
-        std::cerr << "Accepted connection from: " << client_info->h_name << std::endl;
+        std::cerr << "Accepted connection on port " << port << " from: " << client_info->h_name << std::endl;
```

Comment:
This should be a `cout` instead of a `cerr` as it is not an error.

Reply:
It's not an error, but it's not a detail that you would want to know unless you're debugging/monitoring, which is what the error stream is for, no?

Reply:
That is basically the logic I was going for, but happy to change it

Reply:
@KipHamiltons We're running this in Webots, basically the only difference that realistically happens is red text vs black text.
We should reserve the `cerr` for errors as it makes it simpler to see when there is erroneous vs expected behavior.

Reply:
Works for me

Reply:
Good point that this is a webots stream. Fair enough